cmd/spiffe-helper/config.go

```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+	"io/ioutil"
+
+	"github.com/hashicorp/hcl"
+	"github.com/spiffe/spiffe-helper/pkg/sidecar"
+)
+
+// ParseConfig parses the given HCL file into a SidecarConfig struct
+func ParseConfig(file string) (Config *sidecar.Config, err error) {
+	sidecarConfig := &sidecar.Config{
+		AgentAddress:       `hcl:"agentAddress"`,
+		Cmd:                `hcl:"cmd"`,
+		CmdArgs:            `hcl:"cmdArgs"`,
+		CertDir:            `hcl:"certDir"`,
+		SvidFileName:       `hcl:"svidFileName"`,
+		SvidKeyFileName:    `hcl:"svidKeyFileName"`,
+		SvidBundleFileName: `hcl:"svidBundleFileName"`,
+		RenewSignal:        `hcl:"renewSignal"`,
+		Timeout:            `hcl:"timeout"`,
+	}
+
+	// Read HCL file
+	dat, err := ioutil.ReadFile(file)
+	if err != nil {
+		return nil, err
+	}
+
+	hclText := string(dat)
+
+	// Parse HCL
+        hclParseTree, err := hcl.Parse(hclText)
+        if err != nil {
+                return nil, err
+        }
+
+        if err := hcl.DecodeObject(&sidecarConfig, hclParseTree); err != nil {
```

Comment:
sidecarConfig is already a pointer. You don't need to take the address and can just pass it directly. Also the parse and then decodeobject can be simplified to the following:
```suggestion
        if err := hcl.Decode(&sidecarConfig, string(dat)); err != nil {
```

Reply:
done.